jax/experimental/mosaic/gpu/utils.py

```diff
@@ -1115,6 +1124,80 @@ def memref_ptr(memref_arg, memory_space=None):
   )
 
 
+def extern_elementwise_func_apply(func, return_type, arg_list,
+                                  static_arg_idxs=(), pack_elts=False,
+                                  elt_byte_count_idx=None):
+  tidx = thread_idx()
+  wgidx = warpgroup_idx()
+  index = ir.IndexType.get()
+  i64 = ir.IntegerType.get_signless(64)
+  i32 = ir.IntegerType.get_signless(32)
+  ptr_ty = ir.Type.parse("!llvm.ptr")
+  num_elts = None
+  elem_bytewidth = None
+  num_args = len(arg_list)
+  tensor_arg_idxs = set(range(num_args)) - set(static_arg_idxs)
```

Comment:
Can't we infer the "static args" from their types? You can just check for `ir.MemRefType.isinstance` or `ir.ShapedType.isinstance` if for some reason you want to handle tensors/vectors/etc.